lib/check-for-ami.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+let assert = require('assert');
+
+/**
+ * Check for the existence of an AMI that's executable by
+ * the provided ec2 object in the given region
+ */
+
+let debug = require('debug')('aws-provisioner:lib:check-for-ami');
+
+module.exports = async function (ec2, ami, region) {
+  let request = {
+    /* Not sure why this isn't working
+    ExecutableUsers: [
+      'self',
+      'all',
+    ],*/
+    ImageIds: [ami],
+  };
+
+  let result;
+  try {
+    // If we're using the multi-region wrapper, we handle things
+    // slightly differently
+    if (ec2.describeImages && ec2.describeImages.inRegion) {
```

Comment:
Do we have a need for both? can't we just assert that you always give `region` and an instance of `multi-region wrapper` ?